java-squid/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -217,11 +220,15 @@ public void visitMethodInvocation(MethodInvocationTree syntaxNode) {
           }
         }
       } else {
-        final List<SymbolicValue> values = programState.peekValues(syntaxNode.arguments().size() + 1);
-        final List<SymbolicValue> argumentValues = values.subList(1, values.size());
-        for (SymbolicValue target : argumentValues) {
-          programState = closeResource(programState, target);
-        }
+        attemptToCloseArguments(syntaxNode.arguments(), 1);
+      }
+    }
+
+    private void attemptToCloseArguments(final Arguments arguments, int stackOffset) {
```

Comment:
Stack for Method invocation: SV that will contain the value returned by the method + values of the arguments
Stack for NewClass: only values of the arguments.
Alternative is to use a boolean (newClass or methodInvocation)